pytorch_lightning/plugins/training_type/tpu_spawn.py

```diff
@@ -225,9 +233,11 @@ def start_training(self, trainer) -> None:
         # todo: precision pluging is call in accelerator setup and should be moved
         if 'XLA_USE_BF16' in os.environ:
             del os.environ["XLA_USE_BF16"]
+        self._close_logger(trainer)
         xmp.spawn(self.new_process, **self.xmp_spawn_kwargs)
 
     def start_testing(self, trainer) -> None:
+        self._close_logger(trainer)
```

Comment:
Why do we do this?

Reply:
@tchaton is it a pickling issue?

Reply:
The main process logger applies a lock on the log_dir. I need to close the logged, so the spawned rank zero process can actually take the control back and start logging. If not, the process will be locked.

Reply:
And it needs to be done for both train, test as users might not always call fit, but test directly with multiple tpu cores.